miner/src/miner.rs

```diff
@@ -393,18 +394,41 @@ impl MinerService for Miner {
 	}
 
 	fn pending_transactions_hashes(&self) -> Vec<H256> {
-		let transaction_queue = self.transaction_queue.lock().unwrap();
-		transaction_queue.pending_hashes()
+		match (self.sealing_enabled.load(atomic::Ordering::Relaxed), self.sealing_work.lock().unwrap().peek_last_ref()) {
```

Comment:
What's the idea here? Return only transactions from the pending block if sealing is enabled, return all transactions from the queue otherwise? Where is this required?